community-operators/rook/rook.v0.1.0.csv.yaml

```diff
@@ -0,0 +1,346 @@
+      name: volumes.rook.io
+      version: v1alpha2
+      displayName: Ceph Volume
+      description: Represents a Ceph volume.
+  installModes:
+  - type: OwnNamespace
+    supported: true
+  - type: SingleNamespace
+    supported: true
+  - type: MultiNamespace
+    supported: true
+  - type: AllNamespaces
+    supported: false
+  install:
+    spec:
+      clusterPermissions:
+      # BEGIN: to run on OC
+      - rules:
+        - verbs:
+          - use
+          apiGroups:
+          - security.openshift.io
+          resources:
+          - securitycontextconstraints
+          resourceNames:
+          - privileged
+        serviceAccountName:
+          - rook-ceph-system
+          - rook-ceph-mgr
+          - default
+          - rook-ceph-osd
+          # not sure how to reflect the namespace here, our sc was written like:
+            # - system:serviceaccount:rook-ceph-system:rook-ceph-system
+            # - system:serviceaccount:rook-ceph:default
+            # - system:serviceaccount:rook-ceph:rook-ceph-mgr
+            # - system:serviceaccount:rook-ceph:rook-ceph-osd
+      # END: to run on OC
+      - rules:
+        - apiGroups:
+          - ""
+          resources:
+          - secrets
+          - pods
+          - pods/log
+          - services
+          - configmaps
+          verbs:
+          - get
+          - list
+          - watch
+          - patch
+          - create
+          - update
+          - delete
+        - apiGroups:
+          - extensions
+          resources:
+          - deployments
+          - daemonsets
+          - replicasets
+          verbs:
+          - get
+          - list
+          - watch
+          - create
+          - update
+          - delete
+        serviceAccountName: rook-ceph-cluster-mgmt
+      - rules:
+        - apiGroups:
+          - ""
+          resources:
+          - pods
+          - nodes
+          - nodes/proxy
+          verbs:
+          - get
+          - list
+          - watch
+        - apiGroups:
+          - ""
+          resources:
+          - events
+          - persistentvolumes
+          - persistentvolumeclaims
+          - endpoints
+          verbs:
+          - get
+          - list
+          - watch
+          - patch
+          - create
+          - update
+          - delete
+        - apiGroups:
+          - storage.k8s.io
+          resources:
+          - storageclasses
+          verbs:
+          - get
+          - list
+          - watch
+        - apiGroups:
+          - batch
+          resources:
+          - jobs
+          verbs:
+          - get
+          - list
+          - watch
+          - create
+          - update
+          - delete
+        - apiGroups:
+          - ceph.rook.io
+          resources:
+          - '*'
+          verbs:
+          - '*'
+        - apiGroups:
+          - rook.io
+          resources:
+          - '*'
+          verbs:
+          - '*'
+        serviceAccountName: rook-ceph-global
```

Comment:
Are these service accounts used by the operator? The deployment doesn't reference this service account

Reply:
The service accounts used by the operator are `rook-ceph-system`, `rook-ceph-mgr`, and `rook-ceph-osd`. This is a very different structure from the original definitions rook has for the cluster roles and SAs. This should likely be `rook-ceph-system`

Reply:
Most CSVs I've seen define one service account with all the permissions and clusterPermissions the operator will need, so I'm assuming the other service accounts should be set as `rook-ceph-system`

Reply:
@SamiSousa Rook currently requires the three different service accounts. One is used by the rook operator, and the other two are for running pods that are started by the operator. All of this was designed to run with least privilege. How can we define the three service accounts in the same CSV? Or are you saying it's not possible?

Reply:
It should be possible, so no need to change it. I'm just making sure the different service accounts are intentional.

Reply:
Question - do you want all the service accounts to have the same permissions? You can have multiple service accounts each with unique permissions.

Reply:
Yes, each of the service accounts has unique permissions.